pkg/kubelet/volumes.go

```diff
@@ -87,11 +91,22 @@ func (vh *volumeHost) NewWrapperUnmounter(volName string, spec volume.Spec, podU
 		// Not found but not an error
 		return nil, nil
 	}
+
 	c, err := plugin.NewUnmounter(spec.Name(), podUID)
-	if err == nil && c == nil {
-		return nil, errUnsupportedVolumeType
+	if err != nil {
+		return nil, err
+	}
+	if c != nil {
+		// return good volume unmounter
+		return c, nil
 	}
-	return c, nil
+
+	// generic volume type error
+	return nil, createVolumeTypeError(volName)
+}
+
+func createVolumeTypeError(volName string) error {
+	return fmt.Errorf("Unable to mount volume or unsupported volume type for volume: %s for plugin: Unknown - check logs or pod describe events for possible root cause", volName)
```

Comment:
> yeah, so what you are seeing here, is the result of two volume types producing errors, if the user was not using PVC/PV then you would only see the error associated with the underlying source volume (i.e. glusterfs type with the true root cause of missing endpoints) ... but with PVC and PV types you also get another type of persistent-claim with no real error other than it fails at the mounter level but the real error comes from the underlying source plugin type.

Gotcha!

> So I guess we could repress the generic persistent-claim error and only show the underlying issue from the source type??

I think that would be right thing to do. Not prefix errors unless they are too cryptic so that the use would not have to parse through a lot of boiler plate to get to the real issue. I would do that in a separate PR though or open an issue so it is not forgotten.

The instance in this PR though should be addressed here.